### PR TITLE
Serialization bug due to version incompatibility between azure-core and msrest

### LINKED ISSUE #761: [BUG] Serialization Error from Azure Core when installing feathr
### Willingness to contribute

Yes. I can contribute a fix for this bug independently.

### Feathr version

0.8.0

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 20.0)**:
- **Python version**:
- **Spark version, if reporting runtime issue**:
azure-core=1.22.1
msrest==0.7.0

### Describe the problem

It seems it happens due to version mismatch of merest which automatically gets upgraded with azure cor

https://github.com/Azure/azure-sdk-for-python/issues/24765

The root cause is msrest version (0.7.0+) is incompatible with azure-core version (1.22.1). Need to adjust either of them.

### Tracking information

_No response_

### Code to reproduce bug

_No response_

### What component(s) does this bug affect?

- [X] `Python Client`: This is the client users use to interact with most of our API. Mostly written in Python.
- [ ] `Computation Engine`: The computation engine that execute the actual feature join and generation work. Mostly in Scala and Spark.
- [ ] `Feature Registry API`: The frontend API layer supports SQL, Purview(Atlas) as storage. The API layer is in Python(FAST API)
- [ ] `Feature Registry Web UI`: The Web UI for feature registry. Written in React

### PR DESCRIPTION
**Pull request by jainr**
## Description
While installing Feathr we pin to specific version of azure-core (1.22.1) due to prior issues on databricks. This version of azure-core only works with msrest (0.6.1) version.
Some environment might have higher version of azure-core pre-installed (Say 1.24.0) which comes with a higher version of 
msrest. As a result we need to pin the version of msrest to 0.6.1 so it works with pinned version of azure-core (1.22.1)

Resolves #761 

## How was this PR tested?
Tested in local environment and made sure right version of msrest gets installed. 
Screen Shot 2022-10-18 at 7 13 10 AM

## Does this PR introduce any user-facing changes?
- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to clarify your proposed changes.

### CODE REVIEW

File: feathr_project/feathr/udf/_preprocessing_pyudf_manager.py [hunk 1]
```diff
@@ -53,6 +53,8 @@ def build_anchor_preprocessing_metadata(anchor_list: List[FeatureAnchor], local_
                 continue
             preprocessing_func = anchor.source.preprocessing
             if preprocessing_func:
+                print("Preprocessing function defined")
```
Comment (blrchen) [inline feathr_project/feathr/udf/_preprocessing_pyudf_manager.py:56]: Should this debug code get removed before merge?

Comment (jainr) [reply feathr_project/feathr/udf/_preprocessing_pyudf_manager.py]: Good catch, fixed. It was left over from main branch in my fork.

**Review by Yuqing-cat — APPROVED** [review]

File: feathr_project/feathr/udf/_preprocessing_pyudf_manager.py (hunk 1 above)
Comment (jainr) [reply feathr_project/feathr/udf/_preprocessing_pyudf_manager.py]: Yes, removed.

**Review by blrchen — APPROVED** [review]

**Comment by Uditanshu1612** [comment]
Hi,
Is the issue is resolved? I'm also facing the same issue while importing a package.
Currently My azure-core is a version of 1.26.0 so while executing the module using this version of azure-core it throws me an error

image


So here by observing the error I get to know that module requires an azure-core of version <=1.22.1. So, if downgrade the version to 1.22.1 or less than that, then I'm facing an error

image


did anyone have the solution how to resolve the error and successfully import the module?

**Comment by Uditanshu1612** [comment]
Hi,
Is the issue is resolved? I'm also facing the same issue while importing a package.
Currently My azure-core is a version of 1.26.0 and msrest is a version of 0.6.21 so while executing the module using this version of azure-core it throws me an error

image


So here by observing the error I get to know that module requires an azure-core of version <=1.22.1. So, if downgrade the version to 1.22.1 or less than that, with mrest = 0.6.1 then I'm facing an error

image


did anyone have the solution how to resolve the error and successfully import the module?

**Comment by jainr** [comment]
I

> Hi, Is the issue is resolved? I'm also facing the same issue while importing a package. Currently My azure-core is a version of 1.26.0 and msrest is a version of 0.6.21 so while executing the module using this version of azure-core it throws me an error
> 
> image
> 
> So here by observing the error I get to know that module requires an azure-core of version <=1.22.1. So, if downgrade the version to 1.22.1 or less than that, with mrest = 0.6.1 then I'm facing an error
> 
> image
> 
> did anyone have the solution how to resolve the error and successfully import the module?

It seems this is a problem with version mismatch between azure-core and azure-cli. See https://github.com/Azure/azure-cli/issues/23267,  I also see you posted it there too.

What version of azure-cli are you using, could you try to downgrade it to <=2.37.0 and see if the issue still persisits?

**Pull request merged**